en/02_Developer_Guides/09_Security/05_Secure_Coding.md

```diff
@@ -660,115 +660,159 @@ This is a recommended option to secure any controller which displays
-director function `forceSSL()`
+## TLS (aka SSL aka HTTPS)
 
-```php
-use SilverStripe\Control\Director;
+Silverstripe CMS recommends the use of TLS (HTTPS) for your application. You can configure this by setting the `ForceSSL` property on the [`CanonicalURLMiddleware`](api:SilverStripe\Control\Middleware\CanonicalURLMiddleware) singleton.
 
-if (!Director::isDev()) {
-    Director::forceSSL();
-}
+```yml
+---
+After: '#canonicalurls'
+---
+SilverStripe\Core\Injector\Injector:
+  SilverStripe\Control\Middleware\CanonicalURLMiddleware:
+    properties:
+      ForceSSL: true
 ```
 
-`forceSSL()` will only take effect in environment types that `CanonicalURLMiddleware` is configured to apply to (by
+will only take effect in environment types that `CanonicalURLMiddleware` is configured to apply to (by
 default, only `LIVE`). To apply this behaviour in all environment types, you'll need to update that configuration:
 
-```php
-use SilverStripe\Control\Director;
-use SilverStripe\Control\Middleware\CanonicalURLMiddleware;
-
-if (!Director::isDev()) {
-    // You can also specify individual environment types
-    CanonicalURLMiddleware::singleton()->setEnabledEnvs(true);
-    Director::forceSSL();
-}
+```yml
+---
+After: '#canonicalurls'
+---
+SilverStripe\Core\Injector\Injector:
+  SilverStripe\Control\Middleware\CanonicalURLMiddleware:
+    properties:
+      # ...
+      EnabledEnvs:
+        - test
+        - dev
 ```
 
 Forcing HTTPS so requires a certificate to be purchased or obtained through a vendor such as
 [lets encrypt](https://letsencrypt.org/) and configured on your web server.
 
 Note that by default enabling SSL will also enable `CanonicalURLMiddleware::forceBasicAuthToSSL` which will detect
 and automatically redirect any requests with basic authentication headers to first be served over HTTPS. You can
-disable this behaviour using `CanonicalURLMiddleware::singleton()->setForceBasicAuthToSSL(false)`, or via Injector
-configuration in YAML.
+disable this behaviour setting the `ForceBasicAuthToSSL` property to `false` in the YAML configuration.
+
+### Using SSL in database connections
+
+In some circumstances, like connecting to a database on a remote host for example, you may wish to enable SSL encryption to ensure the protection of sensitive information and database access credentials.
+You can configure that by setting the following environment variables:
+
+| Name  | Description |
+| ----  | ----------- |
+| `SS_DATABASE_SSL_KEY` | Absolute path to SSL key file (optional - but if set, `SS_DATABASE_SSL_CERT` must also be set) |
+| `SS_DATABASE_SSL_CERT` | Absolute path to SSL certificate file (optional - but if set, `SS_DATABASE_SSL_KEY` must also be set) |
+| `SS_DATABASE_SSL_CA` | Absolute path to SSL Certificate Authority bundle file (optional) |
+| `SS_DATABASE_SSL_CIPHER` | Custom SSL cipher for database connections (optional) |
```

Comment:
Makes sense to move this to be adjacent to the other bit of docs talking about TLS.